packages/ra-ui-materialui/src/input/ReferenceArrayInput.tsx

```diff
@@ -141,19 +143,33 @@ const ReferenceArrayInput: FC<ReferenceArrayInputProps> = ({
         input,
     });
 
+    const listContext = useMemo(
+        () => ({
+            ...controllerProps,
+            setSort: controllerProps.setSortForList,
```

Comment:
Please add a jsDoc explaining why you do some voodoo here

Reply:
Done